context/db_uptunnel.go

```diff
@@ -106,10 +106,9 @@ func RecoverTunnel(tunnelInfo *TunnelInfo) (tunnel *GTPTunnel) {
 		TEID: tunnelInfo.TEID,
 		PDR:  tunnelInfo.PDR,
 	}
-	var nilVal *NodeIDInDB = nil
 	var nilValNode *DataPathNode = nil
 	empty_nodeID := &NodeIDInDB{}
-	if &tunnelInfo.DataPathNodeUPFNodeID != nilVal {
+	if tunnelInfo != nil {
```

Comment:
Should not this be checked be moved before [line 105](https://github.com/omec-project/smf/blob/master/context/db_uptunnel.go#L105) to make sure that `tunnel` can get `tunnelInfo.TEID` and `tunnelInfo.PDR`?

Reply:
you are right

Reply:
done

Reply:
We are returning `tunnel`, which is only defined if `tunnelInfo != nil`. What happens for the "else" case given that `tunnel` is not defined there? Should we move the `return tunnel` inside the if statement and `return nil` for the else case?

Initial (after your changes)
```go
func RecoverTunnel(tunnelInfo *TunnelInfo) (tunnel *GTPTunnel) {
	if tunnelInfo != nil {
		...
	}
	// TBA: recover dst endPoint
	return tunnel
```

Possible way to handle `else` statement
```go
func RecoverTunnel(tunnelInfo *TunnelInfo) (tunnel *GTPTunnel) {
	if tunnelInfo != nil {
		...
		return tunnel
	}
	// TBA: recover dst endPoint
	return nil
```

Reply:
since `tunnel` is named return value, it is initialized to nil so I think it's the same result.
But I can add your suggestion if you think that is more readable.

Reply:
Ok, if `tunnel` is by default initialized to `nil`, then, no need to update the PR. Thanks!